runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -188,29 +186,39 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
             }
         }
 
-        // Linearize this read against a timestamp
-        final long timestamp =
-                rt.getSequencerView()
-                        .nextToken(Collections.singleton(streamID), 0).getToken()
-                        .getTokenValue();
-        log.debug("Access[{}] conflictObj={} version={}", this, conflictObject, timestamp);
+        /**
+         try to perform the underlying access. Trim failures can happen due to fast
+         cp/trim (i.e. after acquiring the stream tail and before the LU read a trim happens)
+         cycles, in that case the operation is retried. The access is best effort and will
+         give up retrying after a couple of times.
+
+         Note: since the system should always guarantee that the latest version of the object
+         is accessible, retrying a specific number of times then giving up might not as desirable
+         as retrying until the object is accessible, but this will be addressed in checkpointing 2.0.
+         **/
+        for (int x = 0; x < rt.getParameters().getTrimRetry(); x++) {
+            // Linearize this read against a timestamp.
+            final long timestamp = rt.getSequencerView()
+                            .nextToken(Collections.singleton(streamID), 0).getToken().getTokenValue();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "streamID"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)